allium/lib/page_writer.py

```
@@ -751,6 +768,189 @@ def get_detail_page_context(relay_set, category, value):
+        rendered_counts = pool.map(_render_contact_batch_mp, page_args)
+        pool.close()
+        pool.join()
+
+        total_time = time.time() - start_time
+        total_files = sum(rendered_counts)
+        relay_set.progress_logger.log(
+            f"contact page generation complete - Generated {len(page_args)} contacts, {total_files} files in {total_time:.2f}s"
+        )
+        if relay_set.progress and total_files:
+            print(f"    🚀 Parallel: {relay_set.mp_workers} workers, {total_time/total_files*1000:.1f}ms/rendered file avg")
+    except Exception as e:
```

Ruff (0.15.5)

[warning] Do not catch blind exception: `Exception`

(BLE001)